monix-reactive/shared/src/main/scala/monix/reactive/internal/consumers/MapTaskConsumer.scala

```diff
@@ -43,7 +45,10 @@ private[reactive] final class MapTaskConsumer[In, R, R2](source: Consumer[In, R]
             try {
               val task = f(value)
               streamErrors = false
-              task.runAsync(cb)
+              self.synchronized {
+                if (!isCancelled)
```

Comment:
I feel like we don't have to synchronize `isCancelled` if it's `volatile` (or it doesn't have to be `volatile` if it's only used in synchronized block)

Reply:
Yeah, I'm synchronizing access across two variables here. Volatile probably is redundant